Issue #5008: Increased coding pitest coverage to 100%

eorloff opened the pull request:
#5008

HiddenFieldCheck.java:278: Removed unused call. It was always called with LPAREN, the actual params are handled by visitToken with PARAMETER_DEF.

MultipleVariableDeclarationsCheck.java:124: Cleaned up method since the AST is sorted. 

RequireThisCheck.java:454: Removed check for method call since these has already been handled by parentType == METHOD_CALL in processIdent prior to handling field.

romani commented:
@eorloff , thanks a lot for PR.

Please review CIs failures, as you propose change in logic of Checks, we will need regression testing of them by our checkstyle-tester tool

eorloff commented:
@romani I have run the regression tool using default settings. Result available here:
https://eorloff.github.io/fba5ae4287a7cd16cd3dbb803bd60460a73d4eeb/reports/diff/index.html

romani commented:
Thanks a lot.

Let's split fix into two PRs.
In this PR please keep only simple fixes( that are not `Integer.signum`). We will merge it quickly.

Please create second PR for fixes by `Integer.signum` , we need to think about it a bit more, it is serious damage of readability, I need more time to think about it and discuss it.

eorloff commented:
@romani I have removed the things with signum, and will take a look at doing this in a more readable fashion. 

Updated regression: https://eorloff.github.io/f8596b6be6cdaea567023ecdde6fc245f275627f/diff/index.html

eorloff commented:
@rnveach I have updated the percentage in the pom to 100 which is what pitest reports due to rounding. 
The commit message has been updated.
